GhostText.py

```diff
@@ -91,10 +91,17 @@ class ReplaceContentCommand(TextCommand):
     """
 
     def run(self, edit, **args):
-        self.view.replace(edit, sublime.Region(0, self.view.size()), args['text'])
+        # Temporarily disable conversion of tabs to spaces before
+        # pasting text, so that the selections stay valid
+        old_setting = self.view.settings().get('translate_tabs_to_spaces')
+        try:
+            self.view.settings().set('translate_tabs_to_spaces', False)
+            self.view.replace(edit, sublime.Region(0, self.view.size()), args['text'])
+        finally:
+            self.view.settings().set('translate_tabs_to_spaces', old_setting)
```

Comment:
I don't know how ST config works. What happens if there was no specific config set, does this *enforce* it? I don't want to change the user config. 

Is this setting necessary? I'm surprised this was ever an issue

Reply:
The default value for `translate_tabs_to_spaces` in `Default/Preferences.sublime-settings` is `false`.

I ran into the issue because in my `User/Preferences.sublime-settings` I set it to `true`, so it is enabled when creating a new buffer.

In the example on https://ghosttext.fregante.com/troubleshooting/ the text area contains tabs, and they would be translated to spaces after the `replace()` operation, so the positions would be offset and the wrong part of the text was selected.

I found this issue where a similar problem was mentioned, with the suggestion to temporarily override the setting, which fixed the issue: https://github.com/sublimehq/sublime_text/issues/6157

What do you think?

Reply:
To reply to your question directly, my understanding is that the settings are attached to the view, and are initialized based on the default and user settings, so there would always be a value.